### ISSUE TITLE
@wordpress/react-native-bridge

### ISSUE THREAD

**Issue by MuflahNasir** [WordPress/gutenberg#59939]
Currently, this package is not available on npm. And `@wordpress/compose` uses this package in one of its hooks. So when I installed `@wordpress/compose` I got this error while running a mobile app.

`Unable to resolve "@wordpress/react-native-bridge" from "../../node_modules/@wordpress/compose/src/hooks/use-network-connectivity/index.native.js"`.

So when are you going to publish this package to npm? And what is the temporary workaround for this problem?

**Comment by Mamaduka** [comment]
cc @WordPress/native-mobile

**Comment by MuflahNasir** [comment]
is this an npm package?

**Comment by Mamaduka** [comment]
@MuflahNasir, it looks like it's [unpublished package](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-react-native-bridge/), but that's all the information I've. I pinged the team working on the mobile apps.

**Comment by derekblank** [comment]
@MuflahNasir Correct, `@wordpress/react-native-bridge` is not currently published to npm. 

The temporary workaround would be to install the package in your project to use locally:

```
npm install ./gutenberg/packages/react-native-bridge --save
```

There are two hooks in `@wordpress/compose` that use `@wordpress/react-native-bridge` as a dependency:

- [useNetworkConnectivity](https://github.com/WordPress/gutenberg/blob/72b9ffa48a265bdd3c7963862863f61a24e87cad/packages/compose/src/hooks/use-network-connectivity/index.native.js#L8)
- [usePreferredColorScheme](https://github.com/WordPress/gutenberg/blob/72b9ffa48a265bdd3c7963862863f61a24e87cad/packages/compose/src/hooks/use-preferred-color-scheme/index.android.js#L8)

I am not certain if there are plans to publish `@wordpress/react-native-bridge`. Perhaps it may be better to remove these two hooks from `@wordpress/compose` and relocate them somewhere else. We can continue the discussion with the rest of the native mobile team on this thread.

**Comment by fluiddot** [comment]
> I am not certain if there are plans to publish `@wordpress/react-native-bridge`. Perhaps it may be better to remove these two hooks from `@wordpress/compose` and relocate them somewhere else. We can continue the discussion with the rest of the native mobile team on this thread.

Seems that both hooks [`useNetworkConnectivity`](https://github.com/WordPress/gutenberg/blob/72b9ffa48a265bdd3c7963862863f61a24e87cad/packages/compose/src/hooks/use-network-connectivity/index.native.js#L8) and [`usePreferredColorScheme`](https://github.com/WordPress/gutenberg/blob/72b9ffa48a265bdd3c7963862863f61a24e87cad/packages/compose/src/hooks/use-preferred-color-scheme/index.android.js#L8) are not referenced in [the documentation](https://github.com/WordPress/gutenberg/blob/trunk/packages/compose/README.md) and the package `react-native-bridge` is [not declared as a dependency in `compose`](https://github.com/WordPress/gutenberg/blob/19a46ddd4133f5eda3b266e2f002a4beff2d9adf/packages/compose/package.json#L31-L45). Taking this into consideration, plus the fact that the hooks are specific to the mobile native version, makes me wonder if the hooks should be relocated in a different package.

**Comment by dcalhoun** [comment]
I agree that relocating the two Hooks in question is likely the best approach. The current setup also results in a cyclic dependency via the `react-native-aztec` package. Relocating the Hooks might also address that cycle.

**Comment by derekblank** [comment]
Sounds good. I'll work on moving these hooks somewhere more suitable.

**Comment by derekblank** [comment]
The `wordpress/@editor` package seemed like the most obvious package to relocate `useNetworkConnectivity` to, as that package is currently the only place where the hook is used. I've created a draft PR for that change in https://github.com/WordPress/gutenberg/pull/59982, but am open to feedback on the structure.

`usePreferredColor` (and `usePreferredColorScheme`) are used in several other packages, however. Any thoughts on a good candidate to relocate this hook to?

**Comment by fluiddot** [comment]
> The `wordpress/@editor` package seemed like the most obvious package to relocate `useNetworkConnectivity` to, as that package is currently the only place where the hook is used. I've created a draft PR for that change in #59982, but am open to feedback on the structure.

If the hooks are specific to the mobile native version and they require the module `react-native-bridge`, I wonder if they should be relocated to `react-native-editor`. My rationale is that `react-native-editor` is the central point of editor-level logic specific to the mobile native version. WDYT?

**Comment by derekblank** [comment]
> I wonder if they should be relocated to react-native-editor.

Good suggestion. I'll check the structure of `react-native-editor` to see how we might relocate these hooks to that package.

**Comment by MuflahNasir** [comment]
Do we have any updates about this issue. I am getting this issue on Android too.

`Unable to resolve "@wordpress/react-native-bridge" from "../../node_modules/@wordpress/compose/src/hooks/use-preferred-color-scheme/index.android.js"`

**Comment by dcalhoun** [comment]
@MuflahNasir there are no updates at this time. https://github.com/WordPress/gutenberg/pull/59982 was drafted to address the issue, but has not progressed to approval or merge. 

Currently, there are fewer individuals actively contributing to the native mobile editor. I'm unsure of when we might finish addressing this. If I am able to find time, I will try to move things forward.

**Comment by tyxla** [comment]
This is no longer relevant after the React Native implementation was removed in #78747. Closing.
